versioning_base.py

```diff
@@ -1092,8 +1121,80 @@ def add_branch( pg_conn_info, schema, branch, commit_msg,
     pcur.commit()
     pcur.close()
 
+def diff_rev_view_str(pg_conn_info, schema, table, branch, rev_begin, rev_end):
+    """DIFFerence_REVision_VIEW_STRing
+    Create the SQL view string of the specified revision difference (comparison).
+    """
+    pcur = Db(psycopg2.connect(pg_conn_info))
+
+    pcur.execute("SELECT * FROM "+schema+".revisions "
+        "WHERE branch = '"+branch+"'")
+    if not pcur.fetchone():
+        pcur.close()
+        raise RuntimeError("Branch "+branch+" doesn't exist")
+    pcur.execute("SELECT MAX(rev) FROM "+schema+".revisions")
+    [max_rev] = pcur.fetchone()
+    if int(rev_begin) > max_rev or int(rev_begin) <= 0:
+        pcur.close()
+        raise RuntimeError("Revision 1 (begin) "+str(rev_begin)+" doesn't exist")
+    if int(rev_end) > max_rev or int(rev_end) <= 0:
+        pcur.close()
+        raise RuntimeError("Revision 2 (end) "+str(rev_end)+" doesn't exist")
+
+    select_str = ("SELECT "
+    "CASE WHEN "
+        +schema+"."+table+"."+branch+"_rev_begin > "+str(rev_begin)+ " "
+        "AND " +schema+"."+table+"."+branch+"_rev_begin <= "+str(rev_end)+ " "
+        "AND " +schema+"."+table+"."+branch+"_parent IS NULL THEN 'a' "
+    "WHEN (" +schema+"."+table+"."+branch+"_rev_begin > "+str(rev_begin)+ " "
+        "AND " +schema+"."+table+"."+branch+"_rev_end IS NULL "
+        "AND " +schema+"."+table+"."+branch+"_parent IS NOT NULL) "
+        "OR ("+schema+"."+table+"."+branch+"_rev_end >= "+str(rev_end)+" "
+        "AND "+schema+"."+table+"."+branch+"_child IS NOT NULL) "
+        "THEN 'u' "
+    "WHEN " +schema+"."+table+"."+branch+"_rev_end > "+str(rev_begin)+ " "
+        "AND " +schema+"."+table+"."+branch+"_rev_end < "+str(rev_end)+ " "
+        "AND " +schema+"."+table+"."+branch+"_child IS NULL THEN 'd' ELSE 'i' END "
+    "as diff_status, * FROM "+schema+"."+table+ " "
+    "WHERE (" +schema+"."+table+"."+branch+"_rev_begin > "+str(rev_begin)+ " "
+        "AND " +schema+"."+table+"."+branch+"_rev_begin <= "+str(rev_end)+") "
+        "OR (" +schema+"."+table+"."+branch+"_rev_end > "+str(rev_begin)+ " "
+        "AND " +schema+"."+table+"."+branch+"_rev_end <= "+str(rev_end)+ " )")
```

Comment:
I think I recall a case where the "CASE WHEN" was a lot slower then the equivalent "UNION".